### DISCUSSION TITLE
Enable Configuration via Environment Variables

### DISCUSSION THREAD

**Discussion by mscott3** [coreybutler/nvm-windows#625] [Ideas & Feature Requests] [answered]
Other tools in the NPM stack allow for settings in Environment Variables.  We would like this capability in NVM as well.
We think the simplest approach would be to use Go's environment variable expansion feature where the settings file is read ( line 728 of nvm .go):

     ...
     for _, line := range lines {
         line = strings.Trim(line, " \r\n")
     // new code
        line = os.ExpandEnv(line)
     //end new code
         if strings.HasPrefix(line, "root:") {
     ...

This will allow referrning to environment variables in the settings.txt file, e.g.:
    npm_mirror:${NVM_CONFIG_NODE_MIRROR}

**Comment by coreybutler** [comment]
First, this is an interesting idea and I think there is a place for it in the stack. However; I don't think that place is in NVM4W. I am working on NVM4W's successor (project "rt"), which I think would be a more appropriate fit. I wrote a little bit about this effort in the [welcome message](https://github.com/coreybutler/nvm-windows/discussions/565#discussioncomment-58112).

What it really comes down to is this would be a new feature, which should be in a new minor/major release. The upcoming release of 1.1.8 is probably going to be the last, since I'm trying to concentrate efforts on rt.

That said, I encourage discussion around this. It helps to know how many people this would benefit.

**Reply by mhscott64** [reply]
We have submitted PR #632 that addresses this issue.

**Comment by mhscott64** [comment]
We have an infrastructure change at work, and will require this change to continue using NVM for Windows.

**Comment by coreybutler** [comment, marked as answer]
This was merged and will be a part of 1.1.8.
